### ISSUE TITLE
dependency() should have an option to override globally defined paths

### ISSUE THREAD

**Issue by themightyoarfish** [mesonbuild/meson#10910]
**Describe the bug**
`dependency()` cannot be told to look for a dependency in a specific prefix. 

**To Reproduce**
Assume you have multiple versions of a library installed in different installation prefixes. During development you might want to switch between them (e.g. debug and release builds). But you can only set `cmake_prefix_path` and the like globally, there is no way I can see to override this path for a single dependency. The `default_options` argument to `dependency()` cannot be used to override `cmake_prefix_path`.

```
project('foo', 'cpp',
  version : '0.1',
  default_options : ['warning_level=3',
                     'cpp_std=c++17',
                     'cmake_prefix_path=/custom/prefix1',
                     'pkg_config_path=/custom/prefix1/lib/pkgconfig'])

cv2 = dependency('OpenCV', version : '>=4.0.0', method : 'cmake')
pcl = dependency('PCL', version : '>=1.12', components : [ 'common' ])

exe = executable('foo', 'foo.cpp',
  dependencies : [ cv2, pcl ])
```

**Expected behavior**

An option to modify the environment for the execution of the function should be available. The doc sounds like `default_options` should be for this purpose, but it seems I am misunderstanding it

**system parameters**
* native
* macOs 12.4
* Python 3.8.8
* Meson verion 0.63.0
* Ninja 1.11.0

**Comment by eli-schwartz** [comment]
Why can't you just use the global path for this? If you have multiple versions of a dependency installed to different install paths, then I would assume those dependencies are isolated on their own. Adding that path to the global path will only affect one dependency anyway.

**Comment by themightyoarfish** [comment]
If I have the same library version installed once in debug build and once in release build, how do I configure to use either one? I can only set the global path, which has the side effect of influencing other dependencies then.

**Comment by eli-schwartz** [comment]
How does it influence other dependencies?

**Comment by themightyoarfish** [comment]
Because the global path influences where they are found, in case of multiple versions.

**Comment by themightyoarfish** [comment]
Let me try again:

- Dependency A debug build is in prefix `prefixDebug/`
- Dependency A release build is in prefix `prefixRelease/`
- Dependency B debug build is in prefix `prefixDebug/`
- Dependency B release build is in prefix `prefixRelease/`
I want meson to find dependency A's debug build, but dependency B's release build.

CMake has the `PATHS` and `HINTS` argument to `find_package()` for this.

**Comment by eli-schwartz** [comment]
And how do you toggle between these without editing CMakeLists.txt?

Wouldn't it make more sense to have:

> * Dependency A debug build is in prefix `prefixADebug/`
> 
> * Dependency A release build is in prefix `prefixARelease/`
> 
> * Dependency B debug build is in prefix `prefixBDebug/`
> 
> * Dependency B release build is in prefix `prefixBRelease/`

Since prefixDebug/ is not intended to be a general environment for finding multiple software components, it seems odd to *store* multiple software components in it.

**Comment by eli-schwartz** [comment]
Anyway: https://mesonbuild.com/Dependencies.html#cmake

You can set cmake_module_path inside the `dependency()` kwargs, which corresponds to CMake's `-DCMAKE_MODULE_PATH` but not to `-DCMAKE_PREFIX_PATH`. You will unfortunately then have to edit meson.build just like you'd have to edit CMakeLists.txt, which I would say is suboptimal, but if you really want it...

**Comment by themightyoarfish** [comment]
I need to set the prefix path, not the module path.
